core/src/main/java/com/linecorp/armeria/server/annotation/RequestObject.java

```diff
@@ -22,18 +22,51 @@
 import java.lang.annotation.Target;
 
 /**
- * Specifies which parameter should be converted by {@link RequestConverterFunction}.
+ * Specifies which element should be converted by {@link RequestConverterFunction}.
  *
+ * <p>A {@link RequestObject} can be implicitly applied when:
+ * <ul>
+ *     <li>A parameter is neither annotated nor automatically injected type in an annotated method.</li>
+ *     <li>A {@link RequestConverter} annotation is specified on a parameter in an annotated method
+ *     or a setter of a bean.</li>
+ *     <li>A {@link RequestConverter} annotation is specified on a field of a bean.</li>
+ *     <li>A {@link RequestConverter} annotation is specified on a method or constructor which has
+ *     only one parameter.</li>
+ * </ul>
+ * The following example shows when to apply a {@link RequestObject} implicitly.
+ * <pre>{@code
+ * > @RequestConverter(ErinConverter.class)
+ * > @RequestConverter(FrankConverter.class)
+ * > public class CompositeBean {
+ * >     private final Alice a;
+ * >
+ * >     @RequestConverter(BobConverter.class)
+ * >     private Bob b;
+ * >
+ * >     private Charlie c;
+ * >     private David d;
+ * >     private Erin e;
+ * >
+ * >     @RequestObject
+ * >     private Frank f;
```

Comment:
How about adding some comment here why `@RequestObject` has to be specified explicitly?

Reply:
It's hard to explain why, because it must be specified by default. So, I added comments on the element that `@RequestObject` is implicitly applied. What do you think?